tests/test_handler.py

```diff
@@ -0,0 +1,445 @@
+import importlib.machinery
+import importlib.util
+import logging
+import os
+import shutil
+import subprocess
+import textwrap
+import pytest
+from aws_cdk.cx_api import CloudAssembly
+from botocore.credentials import Credentials
+from boto3.session import Session
+from sceptre import exceptions
+from template_handler.cdk import CDK
+from unittest.mock import Mock, MagicMock
+from sceptre.connection_manager import ConnectionManager
+
+name = 'dev/cdk'
+sceptre_user_data = {'object_name': 'object-key.txt'}
+stack_group_config = {'project_code': 'sceptre'}
+arguments = {
+    'path': 'CDK/s3.py'
+}
+aws_profile = 'profile1'
+aws_region = 'eu-west-1'
+aws_iam_role = 'role1'
+
+aws_access_key = '123'
+aws_secret_key = '456'
+aws_token = '789'
+
+connection_manager = MagicMock(
+    spec=ConnectionManager
+)
+
+
+def test_schema():
+    """
+    Test schema function
+    """
+    # Arrange
+    cdk_handler = CDK(
+        name,
+        arguments=arguments,
+        sceptre_user_data=sceptre_user_data,
+        connection_manager=connection_manager,
+        stack_group_config=stack_group_config)
+
+    # Act
+    result = cdk_handler.schema()
+
+    # Assert
+    assert result['type'] == 'object'
+    assert result['properties'] is not None
+    assert result['required'] is not None
+
+
+def test_get_envs():
+    """
+    Test _get_envs function
+    """
+    # Arrange
+    credentials = Credentials(
+        access_key=aws_access_key,
+        secret_key=aws_secret_key
+    )
+
+    connection_manager = MagicMock(
+        spec=ConnectionManager,
+        profile=aws_profile,
+        region=aws_region,
+        iam_role=aws_iam_role
+    )
+
+    connection_manager._get_session = MagicMock(
+        spec_set=Session
+    )
+    connection_manager._get_session.return_value.get_credentials.return_value = credentials
+
+    cdk_handler = CDK(
+        name,
+        arguments=arguments,
+        sceptre_user_data=sceptre_user_data,
+        connection_manager=connection_manager,
+        stack_group_config=stack_group_config
+    )
+
+    # Act
+    result = cdk_handler._get_envs()
```

Comment:
It's generally considered bad form to test private methods. It makes your tests more brittle and could cause them to break in the future even with nonbreaking future changes. Ultimately, your tests should be on the sole public method (`handle`) and your assertions ought to be off the arguments passed to subprocess and the value returned, since those are the boundaries of your unit under test. This test, though, wouldn't be so hard to transform that way, since `env` is an argument you're passing to subprocess anyway.